docs/se/guides/migration_3x.adoc

```diff
@@ -0,0 +1,266 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2022 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon SE 3.x Upgrade Guide
+:description: Helidon SE 3.x Upgrade Guide
+:keywords: helidon, porting, migration, upgrade, incompatibilities
+:rootdir: {docdir}/../..
+
+In Helidon 3 we have made some changes to APIs and runtime behavior. This guide
+will help you upgrade a Helidon SE 2.x application to 3.x.
+
+
+== Java 17 Runtime
+
+Java 11 is no longer supported in Helidon 3. Java 17 or newer is required. Please follow the instructions in xref:{rootdir}/about/prerequisites.adoc[Prerequisites] for proper installation.
+
+== New Routing
+
+Handling routes based on the protocol version is now possible by registering specific routes
+on routing builder.
+
+For further information check xref:../webserver.adoc[WebServer Documentation]
+
+== Http/2 Support
+
+Helidon support of Http/2 is no longer experimental.
+
+Http/2 needed to be explicitly enabled by configuration in Helidon 2:
```

Comment:
HTTP/2 had to be explicitly enabled by configuration in Helidon 2.x.

Reply:
fixed